cpp/src/unary/math_ops.cu

```diff
@@ -234,6 +234,16 @@ struct DeviceNot {
   }
 };
 
+// negation
+
+struct DeviceNegate {
+  template <typename T>
+  __device__ T operator()(T data)
+  {
+    return -data;
```

Comment:
should we reject it for types like boolean and unsigned integer types?

Reply:
Yes thanks. I should use `std::enable_if`, right?